### PR TITLE
feat(quantization): Add warnings when `ObservedGraphModule` does not run a forward pass

### LINKED ISSUE #88609: quantization convert should warn the user if calibration has not happened
### 🐛 Describe the bug

It's an easy mistake to pass the wrong model into `prepare_fx`, such as

```
import copy
import torch
import torch.ao.quantize_fx as quantize_fx

m = M(...)
mp = prepare_fx(copy.deepcopy(m), ...)
# wrong model used for calibration
calibrate(m, data_loader)
mq = convert_fx(mp)
evaluate(mq, data_loader)
```

Currently there is no way for user to realize the mistake until after they evaluated the accuracy, which can take awhile depending on model/dataset.  If the user is doing static quantization, we should warn the user during convert if any activation observers have not seen any data.

### Versions

master

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @jgong5 @Xia-Weiwen @leslie-fang-intel

### PR DESCRIPTION
**Pull request by jon-chuang**
Fixes https://github.com/pytorch/pytorch/issues/88609

Questions:
- do we need to extend this to e.g. `convert_standalone_module`? 
   - this PR seems good as a first pass, but we may be able to catch more cases where users try to convert an observed module that is not a `GraphModule` before running any forward passes through it. 
   - Also not sure about edge cases where the top-level model `forward` pass is not run, but forward passes of observed submodules are... - seems not worth considering

cc @jerryzh168 @jianyuh @raghuramank100 @jamesr66a @vkuzo @jgong5 @Xia-Weiwen @leslie-fang-intel @ezyang @SherlockNoMad @EikanWang @wenzhe-nrv

### CODE REVIEW

File: torch/ao/quantization/fx/graph_module.py [hunk 1]
```diff
@@ -53,6 +55,25 @@ def __deepcopy__(self, memo):
         fake_mod = torch.nn.Module()
         fake_mod.__dict__ = copy.deepcopy(self.__dict__)
         return ObservedGraphModule(fake_mod, copy.deepcopy(self.graph), copy.deepcopy(self.preserved_attr_names))
+    
+    @override
+    def recompile(self) -> PythonCode:
```
Comment (jon-chuang) [inline torch/ao/quantization/fx/graph_module.py:60]: Hmm shouldn't we also ensure we wrap during init as well...?

**Comment by jerryzh168** [comment]
thanks for the PR, we are planning to remove `ObservedGraphModule` and `QuantizedGraphModule` actually, see https://github.com/pytorch/pytorch/pull/92842
